<?xml version="1.0" encoding="UTF-8"?>
<launch>

<!-- Launch Segment table node to remove plane -->
  <arg name="rate"                      default="1"                               />
  <arg name="cloud"                     default="/ext_camera/depth/color/points"  />
  <arg name="segm_cloud"                 default="/segment_table/nonplane"  />
  <arg name="frame"                     default="/ext_camera_link"                 />
  <arg name="passthrough_axis"          default="y"                               />
  <arg name="passthrough_min"           default="-0.6"                             /> <!-- -0.2 a 1 -->
  <arg name="passthrough_max"           default="0.15"                             />
  <arg name="downsampling_size"         default="0.01"                            />
  <arg name="show_rviz"                 default="true"                            />

  <node name="segment_table" pkg="tiago_pcl_tutorial" type="segment_table" output="screen">
    <remap from="cloud"                     to="$(arg cloud)"     />
    <remap from="segmcloud"                 to="$(arg segm_cloud)"     />
    <param name="rate"                      type="double" value="$(arg rate)"  />
    <param name="frame"                     type="string" value="$(arg frame)"  />
    <param name="passthrough_axis"          type="string" value="$(arg passthrough_axis)"  />
    <param name="passthrough_min"           type="double" value="$(arg passthrough_min)"   />
    <param name="passthrough_max"           type="double" value="$(arg passthrough_max)"   />
    <param name="downsampling_size"         type="double" value="$(arg downsampling_size)" />
  </node>

<!-- Launch pcl regions to divide in clusters -->
  <arg name="in_cloud"                     default="/segment_table/nonplane"  />
  <node name="pcl_regions" pkg="tiago_pcl_tutorial" type="region_segment" output="screen">
    <remap from="input_cloud" to="$(arg in_cloud)"/>
  </node>

<!-- Launch RVIZ and Reconfigure -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find tiago_pcl_tutorial)/config/pcl_regions.rviz"/>
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />

</launch>
